modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -871,9 +871,16 @@ def get_objects_from_partitions(cls, partitions):
             assert all(
                 [len(partition.list_of_blocks) == 1 for partition in partitions]
             ), "Implementation assumes that each partition contains a single block."
-            return cls._execution_wrapper.materialize(
-                [partition.list_of_blocks[0] for partition in partitions]
-            )
+            blocks_to_materialize = [
+                partition.list_of_blocks[0] for partition in partitions
+            ]
+            try:
+                result = cls._execution_wrapper.materialize(blocks_to_materialize)
+            except Exception:
+                # wait for the end of computations so that the processes free up the resources used
+                cls._execution_wrapper.wait(blocks_to_materialize)
+                raise
+            return result
```

Comment:
if that code is to stay, we can make it look nicer:
```suggestion
            try:
                return cls._execution_wrapper.materialize(blocks_to_materialize)
            except Exception:
                # wait for the end of computations so that the processes free up the resources used
                cls._execution_wrapper.wait(blocks_to_materialize)
                raise
```